src/parser/parse.ts

```diff
@@ -191,14 +192,24 @@ export class Parser<T extends ParserInput, TFlags extends OutputFlags<T['flags']
           flags[token.flag] = true
         }
 
-        // eslint-disable-next-line no-await-in-loop
-        flags[token.flag] = await flag.parse(flags[token.flag], this.context, flag)
+        try {
```

Comment:
Could you abstract this out to a function so you don't have to repeat yourself here and on line 206? For instance,

```typescript
function parse(name: string, context: any, flag: BooleanFlag<any> | CompletableOptionFlag<any>) {
  try {
    // return await flag.parse ...
  } catch {
    // throw error
  }
}
```

Reply:
Absolutely! Just pushed a new commit that uses a new private function 👍